src/components/blogs/blogs.tsx

```diff
@@ -97,37 +98,38 @@ export function SingleBlog(props: {id: string}) {
      */
     useEffect(() => {
         const blogWrapper = document.getElementsByClassName("single-blog-wrapper")[0] as HTMLElement;
-        if (blogWrapper.childNodes.length !== 0) blogWrapper.style.height = "auto";
+        if (blogWrapper.childNodes.length !== 0) {
+            blogWrapper.style.height = "auto";
+            scrollToAnchor();
+        }
     }, [mdWrapper]);
 
     return (
         <>
             <ReactMarkdown
                 children={text}
-                remarkPlugins={[remarkGfm, remarkMath]}
-                rehypePlugins={[rehypeHighlight, rehypeRaw, rehypeKatex]}
-                components={{
-                    h1: HeadingRenderer,
-                    h2: HeadingRenderer,
-                    h3: HeadingRenderer,
-                    h4: HeadingRenderer,
-                    h5: HeadingRenderer,
-                    h6: HeadingRenderer
-                }}
+                remarkPlugins={[
+                  remarkGfm, remarkMath,
+                ]}
+                rehypePlugins={[
+                  rehypeHighlight, rehypeRaw, rehypeKatex, rehypeSlug,
+                ]}
             />
         </>
     );
 }
 
-function HeadingRenderer(props: any) {
-    var children = React.Children.toArray(props.children)
-    var text = children.reduce(flatten, '')
-    var slug = text.toLowerCase().replace(/\W/g, '-')
-    return React.createElement('h' + props.level, {id: slug}, props.children)
+function scrollToAnchor() {
+    const target = window.location.hash;
+    if (!target) return;
+    const anchor = document.querySelector(target) as HTMLElement;
+    if (anchor) {
+      setTimeout(() => {
```

Comment:
There was an issue when I went to a specific URL in the blog as "https://localhost:3000/blogs/a-blog#a-heading", it didn't go to the correct heading. The reason is most likely the document wasn't loaded in order, so I set a small timeout here to make sure the document loaded completely and then perform scrollTo